### ISSUE TITLE
Newsletter not sent to all members of group

### ISSUE THREAD

**Issue by CWalkden** [tendenci/tendenci#884]
Symptom:
I send a newsletter to a group with 334 members.
The confirmation text says that 42 emails were sent.

Background:
We had some users in this group already.  Probably around 42.
We imported users into the group from a .csv file with just Username, Firstname, Lastname and email address.  About 290 users.

So it seems that the newly imported users are not receiving emails.

First thing we found was that the direct_mail field was set to false (this was blank in the .csv import file).  We used database tools to set this to true for all of the users we imported.  A test resulted in 42 emails again.
Then I tried to reproduce the problem by creating a test group, and importing two new users into this group.  This works fine, both are emailed when I send a newsletter.
Next I tried adding one of the previously imported users into the test group.  I'm guessing that this user was not one of the 42 who were successful last time, but I can't check this.  A newsletter sent to this group successfully sends three emails.

Any advice on what might be happening, or what we could do to track down this problem further?  It is particularly difficult to test, since we don't want to spam those 42 users with test emails!

**Comment by CWalkden** [comment]
Another clue:
When viewing the group, I selected "Export all 334 members".
When the file was ready, it said "42 members exported".
There are 334 records in the downloaded file.

**Comment by jennyq** [comment]
@CWalkden,

Where did it tell you "42 members exported"?

In your explorer playground /explorer/play/, you can enter this SQL (Note, replace 1 with your goup id for the last line). Check how many rows you get and if they all have valid email addresses. 

    select u.email, u.first_name, u.last_name, gm.id, gm.group_id
    from user_groups_groupmembership gm
    inner join auth_user u on u.id=gm.member_id
    where gm.status_detail='active'
    and gm.status=True
    and gm.is_newsletter_subscribed=True
    and gm.group_id=1;

**Comment by CWalkden** [comment]
Thanks @jennyq, the "valid email addresses" was the clincher!
I found that the .csv file we had used to import the users contained an extra space after the email address.  For example "joeblogs@gmail.com ".
I was mistaken thinking that it had told me there were 42 members exported.  What it actually told me was that it had exported the users from group 42.  42 was in the filename of the exported .csv file, and I had somehow decided this was the number of group members  Probably because 42 was also the number of emails that the newsletters module had reported sent.

I was able to correct the email addresses in the .csv file, then upload it again, overwriting the previously imported users.

It was interesting that the Newsletter module reported that it did not send emails to these addresses.  This suggests that it knew they were not valid in some way.  It would be helpful if it could report on which users were skipped.

It would also be useful if the user import process could remove leading and trailing spaces.
